src/Aspire.Hosting.Azure.AIFoundry/AzureAIFoundryLocalResourceExtensions.cs

```diff
@@ -0,0 +1,259 @@
+
+                await rns.PublishUpdateAsync(resource, state => state with
+                {
+                    State = new ResourceStateSnapshot(KnownResourceStates.Starting, KnownResourceStateStyles.Info)
+                }).ConfigureAwait(false);
+
+                await manager.StartServiceAsync(ct).ConfigureAwait(false);
+
+                resource.ApiKey = manager.ApiKey;
+
+                if (manager.IsServiceRunning)
+                {
+                    if (resource.TryGetLastAnnotation<EndpointAnnotation>(out var endpoint))
+                    {
+                        endpoint.AllocatedEndpoint = new AllocatedEndpoint(
+                            new EndpointAnnotation(
+                                System.Net.Sockets.ProtocolType.Tcp,
+                                manager.Endpoint.Scheme,
+                                "http",
+                                "http",
+                                manager.Endpoint.Port,
+                                manager.Endpoint.Port,
+                                false,
+                                false),
+                            manager.Endpoint.ToString(),
+                            manager.Endpoint.Port);
+                    }
+
+                    await rns.PublishUpdateAsync(resource, state => state with
+                    {
+                        State = KnownResourceStates.Running,
+                        Properties = [.. state.Properties, new(CustomResourceKnownProperties.Source, "Foundry Local")]
+                    }).ConfigureAwait(false);
+                }
+                else
+                {
+                    await rns.PublishUpdateAsync(resource, state => state with
+                    {
+                        State = KnownResourceStates.FailedToStart,
+                        Properties = [.. state.Properties, new(CustomResourceKnownProperties.Source, "Foundry Local")]
+                    }).ConfigureAwait(false);
+                }
+
+            }, ct));
+
+        return builder;
+    }
+
+    /// <summary>
+    /// Ensures that the resource stops when the application is shutting down.
+    /// </summary>
+    /// <param name="resource">The resource builder for the Foundry Local resource.</param>
+    /// <returns>The updated resource builder.</returns>
+    private static IResourceBuilder<AzureAIFoundryResource> EnsureResourceStops(this IResourceBuilder<AzureAIFoundryResource> resource)
+    {
+        resource.ApplicationBuilder.Eventing.Subscribe<ResourceReadyEvent>(resource.Resource, static (@event, ct) =>
+        {
+            _ = Task.Run(async () =>
+            {
+                var rns = @event.Services.GetRequiredService<ResourceNotificationService>();
+                var manager = @event.Services.GetRequiredService<FoundryLocalManager>();
+
+                await rns.WaitForResourceAsync(@event.Resource.Name, KnownResourceStates.Finished, ct).ConfigureAwait(false);
```

Comment:
This doesn't do anything if you are using a custom resource. Who marks the resource as finished?